pkg/queue/prometheus_stats_reporter_test.go

```diff
@@ -78,29 +83,84 @@ func TestNewPrometheusStatsReporter_negative(t *testing.T) {
 			config,
 			revision,
 			"",
+			1 * time.Second,
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			if _, err := NewPrometheusStatsReporter(test.namespace, test.config, test.revision, test.pod); err.Error() != test.result.Error() {
+			if _, err := NewPrometheusStatsReporter(test.namespace, test.config, test.revision, test.pod, test.reporterReportingPeriod); err.Error() != test.result.Error() {
 				t.Errorf("Got error msg from NewPrometheusStatsReporter(): '%+v', wanted '%+v'", err, test.errorMsg)
 			}
 		})
 	}
 }
 
 func TestReporter_ReportNoProxied(t *testing.T) {
-	testReportWithProxiedRequests(t, &autoscaler.Stat{RequestCount: 39, AverageConcurrentRequests: 3}, 39, 3, 0, 0)
+	testReportWithProxiedRequests(t, &autoscaler.Stat{RequestCount: 39, AverageConcurrentRequests: 3}, 39, 3, 0, 0, 1*time.Second)
 }
 
 func TestReporter_Report(t *testing.T) {
-	testReportWithProxiedRequests(t, &autoscaler.Stat{RequestCount: 39, AverageConcurrentRequests: 3, ProxiedRequestCount: 15, AverageProxiedConcurrentRequests: 2}, 39, 3, 15, 2)
+	testReportWithProxiedRequests(t, &autoscaler.Stat{RequestCount: 39, AverageConcurrentRequests: 3, ProxiedRequestCount: 15, AverageProxiedConcurrentRequests: 2}, 39, 3, 15, 2, 1*time.Second)
 }
 
-func testReportWithProxiedRequests(t *testing.T, stat *autoscaler.Stat, reqCount, concurrency, proxiedCount, proxiedConcurrency float64) {
+func TestReporter_WithDifferentReportingPeriods(t *testing.T) {
```

Comment:
Can we rename this to `TestReporterReport` and fold the two tests above into it? I think `TestReporter_Report` is already obsolete with the third test in your new list and `TestReporter_ReportNoProxied` can easily be recreated in your new (better) tabletest as well! We can then also drop the unnecessary `testReportWithProxiedRequests` helper and have a cleaner test overall.

WDYT?

Reply:
Totally agree. I cleaned up the file.